mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -197,7 +197,24 @@ def _download_file(self, remote_file_path, local_path):
             blob.readinto(file)
 
     def delete_artifacts(self, artifact_path=None):
-        raise MlflowException("Not implemented yet")
+        from azure.core.exceptions import ResourceNotFoundError
+        (container, _, dest_path, _) = self.parse_wasbs_uri(self.artifact_uri)
+        container_client = self.client.get_container_client(container)
+        if artifact_path:
+            dest_path = posixpath.join(dest_path, artifact_path)
+
+        try:
+            blobs = container_client.list_blobs(name_starts_with=dest_path)
+            blob_list = list(blobs)
+            if not blob_list:
+                raise MlflowException(f"No such file or directory: '{dest_path}'")
+
+            for blob in blob_list:
+                container_client.delete_blob(blob.name)
+        except ResourceNotFoundError:
+            raise MlflowException(f"No such file or directory: '{dest_path}'")
```

Comment:
When will this be triggered?

Reply:
If the file is deleted externally from the Azure Storage between the `list_blobs` call and the `delete_blob` call, i.e. a race condition. I was able to easily "reproduce it" by calling `delete_blob` twice, which produces this error. Hopefully that makes sense

Reply:
got it make sense